site/profile/manifests/core/icinga_master.pp

```diff
@@ -0,0 +1,398 @@
+      },
+    },
+  }
+  class { '::icinga2::feature::notification':
+    ensure    => present,
+    enable_ha => true,
+  }
+  icinga2::object::apiuser { $api_user:
+    ensure      => present,
+    password    => $api_pwd,
+    permissions => [ '*' ],
+    target      => '/etc/icinga2/features-enabled/api-users.conf',
+  }
+  ##Icinga2 Perfdata
+  class {'::icinga2::feature::perfdata':
+    ensure => present,
+  }
+  file { '/var/lib/pnp4nagios':
+    ensure  => 'directory',
+    owner   => 'icinga',
+    group   => 'icinga',
+    mode    => '0755',
+    require => Package[$packages],
+    notify  => Service['npcd'],
+  }
+  ##IcingaWeb Config
+  class {'::icingaweb2':
+    manage_repo   => false,
+    logging_level => 'INFO',
+    require       => Class['::icinga2'],
+  }
+  class {'icingaweb2::module::monitoring':
+    ensure            => present,
+    ido_host          => $master_ip,
+    ido_type          => 'mysql',
+    ido_db_name       => $mysql_icingaweb_db,
+    ido_db_username   => $mysql_icingaweb_user,
+    ido_db_password   => $mysql_icingaweb_pwd,
+    commandtransports => {
+      $api_name => {
+        transport => 'api',
+        host      => $master_ip,
+        port      => 5665,
+        username  => $api_user,
+        password  => $api_pwd,
+      }
+    }
+  }
+  ##IcingaWeb LDAP Config
+  icingaweb2::config::resource{ $ldap_resource:
+    type         => 'ldap',
+    host         => $ldap_server,
+    port         => 389,
+    ldap_root_dn => $ldap_root,
+    ldap_bind_dn => $ldap_user,
+    ldap_bind_pw => $ldap_pwd,
+  }
+  icingaweb2::config::authmethod { 'ldap-auth':
+    backend                  => 'ldap',
+    resource                 => $ldap_resource,
+    ldap_user_class          => 'inetOrgPerson',
+    ldap_filter              => $ldap_user_filter,
+    ldap_user_name_attribute => 'uid',
+    order                    => '05',
+  }
+  icingaweb2::config::groupbackend { 'ldap-groups':
+    backend                   => 'ldap',
+    resource                  => $ldap_resource,
+    ldap_group_class          => 'groupOfNames',
+    ldap_group_name_attribute => 'cn',
+    ldap_group_filter         => $ldap_group_filter,
+    ldap_base_dn              => $ldap_group_base,
+  }
+  icingaweb2::config::role { 'Admin User':
+    groups      => 'icinga-admins',
+    permissions => '*',
+  }
+  ##IcingaWeb Director
+  class {'icingaweb2::module::director':
+    git_revision  => 'v1.7.2',
+    db_host       => $master_ip,
+    db_name       => $mysql_director_db,
+    db_username   => $mysql_director_user,
+    db_password   => $mysql_director_pwd,
+    import_schema => true,
+    kickstart     => true,
+    endpoint      => $master_fqdn,
+    api_host      => $master_ip,
+    api_port      => 5665,
+    api_username  => $api_user,
+    api_password  => $api_pwd,
+    require       => Mysql::Db[$mysql_director_db],
+  }
+  ##IcingaWeb PNP
+  exec { 'git clone https://github.com/Icinga/icingaweb2-module-pnp.git pnp':
```

Comment:
Use the `Vcsrepo` type instead of directly exec'ing git.  Directly calling `git` will break if the remote changes, there's a loose files in the repo, etc.

Reply:
changed to vcsrepo

Reply:
also, php is being handled by icingaweb2-director, so no need to install it